### PR TITLE
Reorg mock entity state helpers, add `createMockMetadata`

### PR DESCRIPTION
**Pull request by kulyk**
Epic #26960

Makes `createMockEntitiesState` from `metabase-store/mocks` private. Adds extra types to `createEntitiesState` that should force us to update it anytime the `EntitiesStore` type from `metabase-store` is updated. Also adds the `createMockMetadata` helper to `__support__/metadata` that should encapsulate metadata selector usage in tests. It should now be possible to setup a test like that:

```ts
import { createMockMetadata } from "__support__/store"
import { createMockCard } from "metabase-types/api/mocks"
import { createSampleDatabase } from "metabase-types/api/mocks/presets"

const card = createMockCard()

const metadata = createMockMetadata({
  databases: [ createMockSampleDatabase() ],
  cards: [ card ],
})

const question = metadata.question(card.id);

// Also, we can pass options to the underlying selector
// to test behaviors around sensitive data and hidden tables

createMockMetadata(entities, {
  includeHiddenTables: true,
  includeSensitiveFields: true,
})
```

### How to verify

CI should be green 🟢

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30560)

### CODE REVIEW

File: frontend/test/__support__/store.ts [hunk 1]
```diff
@@ -56,7 +48,42 @@ export interface EntitiesStateOpts {
   questions?: Card[];
 }
 
+const EntitiesSchema: Record<keyof EntitiesState, NormalizrSchema<any>> = {
+  actions: [ActionSchema],
+  alerts: [],
```
Comment (kulyk) [inline frontend/test/__support__/store.ts:53]: These types should make TypeScript throw an error if we don't provide an explicit schema for a new entity added to `EntitiesState`. Not every entity needs a normalizr schema, we can just use `[]` in that case as we do with `alerts`

File: frontend/test/__support__/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { createEntitiesState, EntitiesStateOpts } from "./store";
+
+export function createMockMetadata(
+  entities: EntitiesStateOpts,
+  opts?: MetadataSelectorOpts,
+) {
+  const state = createMockState({ entities: createEntitiesState(entities) });
+  const metadata = getMetadata(state, opts);
+  return { metadata, state };
+}
```
Comment (kulyk) [inline frontend/test/__support__/metadata.ts:12]: Not yet sure if giving back state is a good idea, but looking at the `DashCardMenu` test it feels handy. Looking forward for more feedback

Comment (kulyk) [reply frontend/test/__support__/metadata.ts:12]: Would it make sense to add a shorthand for creating metadata for the sample database preset? 🤔

File: frontend/test/__support__/store.ts [hunk 3]
```diff
@@ -56,7 +48,42 @@ export interface EntitiesStateOpts {
...
+  tables: [TableSchema],
+  fields: [FieldSchema],
+  metrics: [MetricSchema],
+  segments: [SegmentSchema],
+  snippets: [SnippetSchema],
+  users: [UserSchema],
+  questions: [QuestionSchema],
+};
+
+const EMPTY_STATE: EntitiesState = {
```
Comment (ranquild) [inline frontend/test/__support__/store.ts:67]: @kulyk what if we exported `getPlaceholderEntitiesState` but make it not accept any parameters? This way we won't have to duplicate the list of empty objects here

**Review by ranquild — APPROVED** [review]
@kulyk could we also rename `createEntitiesState` to `createMockEntitiesState`?

File: frontend/test/__support__/store.ts (hunk 3 above)
Comment (ranquild) [reply frontend/test/__support__/store.ts:67]: And we could leave `getPlaceholderEntitiesState` in its own `entities.ts` file in that case

Comment (kulyk) [reply frontend/test/__support__/store.ts:67]: Oh, that's much better, good call 👍

**Pull request merged**